python/src/opendp/polars.py

```diff
@@ -398,25 +406,46 @@ def wrap(*args, **kwargs):
 
         def resolve(self) -> Measurement:
             """Resolve the query into a measurement."""
-            from opendp.context import PartialChain
 
             # access attributes of self without getting intercepted by Self.__getattribute__
             lf_plan = object.__getattribute__(self, "_lf_plan")
             query = object.__getattribute__(self, "_query")
             input_domain, input_metric = query._chain
+            d_in, d_out = query._d_in, query._d_out
+
+            def make(scale, threshold=None):
+                return make_private_lazyframe(
+                    input_domain=input_domain,
+                    input_metric=input_metric,
+                    output_measure=query._output_measure,
+                    lazyframe=lf_plan,
+                    global_scale=scale,
+                    threshold=threshold,
+                )
+            
+            if query._output_measure.type.origin == "FixedSmoothedMaxDivergence":  # type: ignore[union-attr]
+
+                # search for a scale parameter. If threshold unknown, set it to u32::MAX so as not to interfere
+                scale = binary_search(
+                    lambda s: make(s, 2**32 - 1).map(d_in)[0] < d_out[0],  # type: ignore[index]
```

Comment:
In the line above, `threshold` doesn't mean the parameter here... I think you mean farther down? Consider rephrasing or moving comment for clarity.